### DISCUSSION TITLE
How the ziehen drag library should be implemented

### DISCUSSION THREAD

**Discussion by brandontrabon** [Beehive-Software-Consultants/beesoft-components#74] [Ideas]
Currently I am basing the ziehen drag and drop library on the dragula library. The idea behind doing this is:

1. The last commit to the dragula library was around 3 years ago.
2. Most of the initial work of dragula was around 8 years ago and there are more advancements in JavaScript since that time.
3. There are features I always wished dragula had or would implement.
4. There are some quirks about dragula that I would like to fix.

Dragula was created as a function that allows containers to be passed and also have the ability for the user to tell the library when something is happening by hooking up to functions that fire on certain DOM events. This seems like a very performant way to do things, but as I'm implementing it (and looking through dragula's code) I am seeing that there is a real mess of functions that make it hard to tell what is going on. The question I have is what does everyone think about possibly implementing this as a class library instead to allow us to separate functions into their logical sections. The one problem I could see with doing this is performance, but there is code I see in dragula that doesn't really appear to be the most performant. What are everyone's thoughts @Beehive-Software-Consultants/development-team? How much less performant is JavaScript classes from functions?

**Comment by klivsp** [comment]
Hi Brandon , I've read a bit for differences and apart of performance, functions are good to be used when we have to work with events on javascript, on the other hand classes make the code cleaner .

**Reply by brandontrabon** [reply]
Yeah you sound like me...I'm kind of torn which direction to take. I think I will stay with the functional path I'm on and once I get something working try and clean it up a bit. When I created the formatted input I did that using JavaScript classes (you can find that in beesoft-components if you want to take a look) and it worked relatively well. Still I think performance is a priority with this library, so staying with functions will help in that regard.

**Reply by klivsp** [reply]
It's a bit difficult to choose , maybe we can try functional path at first and see how things would go.
